integrations/GoogleAds/browser.js

```diff
@@ -48,44 +52,95 @@ class GoogleAds {
+      if (conversionData.conversionLabel) {
+        const { conversionLabel } = conversionData;
+        const { eventName } = conversionData;
+        const sendToValue = `${this.conversionId}/${conversionLabel}`;
+        let properties = {};
+        if (rudderElement.message.properties) {
+          properties.value =
+            rudderElement.message.properties.revenue ||
+            rudderElement.message.properties.value;
+          properties.currency = rudderElement.message.properties.currency;
+          properties.transaction_id = rudderElement.message.properties.order_id;
+        }
+        properties.send_to = sendToValue;
+        properties = removeUndefinedAndNullValues(properties);
+        window.gtag("event", eventName, properties);
+      }
+    } else {
+      const { event } = rudderElement.message;
+      if (!event) {
+        logger.error("Event name not present");
+        return;
+      }
+
+      const sendToValue = this.conversionId;
+      let payload = {};
       if (rudderElement.message.properties) {
-        properties.value = rudderElement.message.properties.revenue;
-        properties.currency = rudderElement.message.properties.currency;
-        properties.transaction_id = rudderElement.message.properties.order_id;
+        payload.value =
+          rudderElement.message.properties.revenue ||
+          rudderElement.message.properties.value;
       }
-      properties.send_to = sendToValue;
-      window.gtag("event", eventName, properties);
+
+      let extraFields = {};
+      try {
+        extraFields = extractCustomFields(
+          rudderElement.message,
+          extraFields,
+          ["properties"],
+          ["revenue", "value"]
+        );
+      } catch (err) {
+        logger.debug(`Error occured at extractCustomFields ${err}`);
+      }
+
+      payload = { ...payload, ...extraFields };
+      payload.send_to = sendToValue;
+      payload = removeUndefinedAndNullValues(payload);
+      window.gtag("event", event, payload);
     }
   }
 
   page(rudderElement) {
     logger.debug("in GoogleAdsAnalyticsManager page");
-    const conversionData = this.getConversionData(
-      this.pageLoadConversions,
-      rudderElement.message.name
-    );
-    if (conversionData.conversionLabel) {
-      const { conversionLabel } = conversionData;
-      const { eventName } = conversionData;
-      window.gtag("event", eventName, {
-        send_to: `${this.conversionId}/${conversionLabel}`,
-      });
+    if (!this.dynamicRemarketing) {
+      const conversionData = this.getConversionData(
+        this.pageLoadConversions,
+        rudderElement.message.name
+      );
+      if (conversionData.conversionLabel) {
+        const { conversionLabel } = conversionData;
+        const { eventName } = conversionData;
+        window.gtag("event", eventName, {
+          send_to: `${this.conversionId}/${conversionLabel}`,
+        });
+      }
+    } else {
+      const event = rudderElement.message.name;
```

Comment:
For both of the track and page calls, if dynamic marketing is enabled, you are not checking the presence of "conversionData.conversionLabel", is that not required for dynamic marketing?

Reply:
It is not required. Conversion Label is not used in dynamic re-marketing.